### DISCUSSION TITLE
View contents without uncompressing

### DISCUSSION THREAD

**Discussion by rmsa5** [aonez/Keka#1085] [Ideas]
Isn't there an obvious feature missing in Keka: the ability to view view/explore contents of a compressed archive (whatever the extension: akin to mounting it as a virtual drive), and to optionally select a subset of files / directories to uncompress, rather than having to uncompress the whole archive before accessing its content? The way WinZIP, WinRAR (7zip too?) allow it on Windows. This is especially useful when dealing with (very) large archives where only a few files need to be extracted.

It is quite surprising what is quite universally praised as the best archive utility on MacOS is lacking such an essential feature.

Thank you in advance, and thanks for the awesome work until now.

**Comment by gingerbeardman** [comment]
See #100

**Comment by JonyChuh** [comment]
View contents without uncompressing
The ability to view the contents of a compressed file without decompressing it is very necessary. This allows you to browse the specific files in the compressed package before decompressing, so that you can selectively decompress one, several, or all files as needed, without having to decompress the entire package just to access certain files. This is especially useful and practical when dealing with larger compressed packages.
Hope you can adopt it. Thank you for what you do!

**Comment by aonez** [comment]
You can already do this with the iOS version that can already be installed on 2020 or newer Macs.

**Reply by JonyChuh** [reply]
Okay. But, do I need to buy a new updated Mac in order to use this feature? That sounds stupid.

**Comment by mehmetnadir** [comment]
You should embrace this idea on MacOS. Because I want to see multi-part files without opening them, I want to get only what is useful. In any case, after extracting the whole file (sometimes I have to wait for minutes) I get what I want and delete everything else.

We should have an option to view the file before opening it.

1- Extract the file instantly
2- View the file before extracting it (then it should extract if we give the extract command)
